### ISSUE TITLE
Rounding of stretch/length with dt unit

### ISSUE THREAD

**Issue by itoko** [Qiskit/openqasm#152]
I think the specification should touch on rounding of stretch/length with dt unit.

For example, how should compilers resolve stretch `s` in this case?
```
qubit q[2];
stretch s;
barrier q;
my_gate q[0];  # lengthof(my_gate) == 20dt
delay[3*s] q[1];
barrier q;
```
Should it raise an error (20dt cannot be divided by 3)?

(Another example)
```
qubit q[2];
stretch s;
barrier q;
my_gate q[0];  # lengthof(my_gate) == 20dt
delay[s] q[1];
my_half_gate q[1];  # lengthof(my_half_gate) == 10dt
delay[2*s] q[1];
barrier q;
```

**Comment by zachschoenfeld33** [comment]
I do not think this should be an issue. `20dt/3` will resolve to a floating type ultimately (once the value of `dt` is plugged in from the backend). I also think this would be an issue to raise for a compiler written for OpenQASM3, rather than in the specification itself...
